src/services/refactors/moveToFile.ts

```diff
@@ -1109,4 +1114,45 @@ function isNonVariableTopLevelDeclaration(node: Node): node is NonVariableTopLev
     }
 }
 
+function moveStatementsToTargetFile(changes: textChanges.ChangeTracker, program: Program, statements: readonly Statement[], targetFile: SourceFile, toMove: ToMove) {
+    const checker = program.getTypeChecker();
+    const sourceExports = new Set<TopLevelDeclarationStatement>();
+    for (const node of toMove.all) {
+        if (isTopLevelDeclarationStatement(node) && hasSyntacticModifier(node, ModifierFlags.Export)) {
+            sourceExports.add(node);
+        }
+    }
+
+    let lastReExport: ExportDeclaration | undefined;
+    for (const node of targetFile.statements) {

```

Comment:
I’m not sure, but I wonder if it would be more efficient to start from `targetFile.symbol.exports`

Reply:
I've changed to `targetFile.symbol.exports`

Reply:
What do you think? Does this seem better? Worse? About the same? 😄

Reply:
While there are advantages to operating with precisely exported symbols, we still need to search for export declarations and avoid unnecessary repetition when navigating to the same declarations. This is particularly important in cases like this: `export { a, b, c } from './a'`;

Reply:
I think we should optimize for the case where this new code does nothing—most of the time, move to file won’t need to delete a re-export. By starting from `targetFile.symbol.exports`, I feel like you can now add a fast path to do nothing by comparing string names, before needing to do alias resolution. If we look at Daniel’s test case again where we move `getGreeting` to `basics.ts`:

`greeting.ts`

```ts
export function getHello() {
    return "hello";
}

export function getWorld() {
    return "world";
}

export function getGreeting() {
    return `${getHello()} ${getWorld()}`;
}
```

`basics.ts`

```ts
export function goodbye() {
    return "goodbye!";
}

export {
    getHello,
    getWorld,
    getGreeting
} from "./greeting";
```

I would expect us to be able to do something like this:

1. Collect names that are exported in the code to move: `["getGreeting"]`
2. For each of those, check `!!targetFile.exports.get(exportName)`. If the name isn’t already exported, we don’t need to do anything. If it is, then we need to do alias resolution to see if it refers to the same thing we’re moving.

Does that make sense or am I forgetting something important?